runtime/compiler/control/J9CompilationStrategy.cpp

```diff
@@ -39,95 +39,26 @@
 #include "env/ut_j9jit.h"
 #include "env/CompilerEnv.hpp"
 
-// NOTE: TR::CompilationController is actually defined in control/OptimizationPlan.hpp
-
-TR::CompilationStrategy *TR::CompilationController::_compilationStrategy = NULL;
-TR::CompilationInfo *    TR::CompilationController::_compInfo = 0;
-int32_t                  TR::CompilationController::_verbose = 0;
-bool                     TR::CompilationController::_useController = false;
-bool                     TR::CompilationController::_tlsCompObjCreated = false;
-
-
-//------------------------------------ init -----------------------------------
-// Initializes the compilationController.
-// Return false if it fails
-//-----------------------------------------------------------------------------
-bool TR::CompilationController::init(TR::CompilationInfo *compInfo)
-   {
-   _useController = false; // Default to failure
-   _compilationStrategy = 0; // Default to failure
-   TR::Options *options = TR::Options::getCmdLineOptions();
-   char *strategyName = options->getCompilationStrategyName();
-
-   if (strategyName && strcmp(strategyName, "none"))
-      {
-      _compInfo = compInfo;
-      if (strcmp(strategyName, "default") == 0)
-         _compilationStrategy = new (PERSISTENT_NEW) TR::DefaultCompilationStrategy();
-      else if (strcmp(strategyName, "threshold") == 0)
-         _compilationStrategy = new (PERSISTENT_NEW) TR::ThresholdCompilationStrategy();
-      else // if no match, use default
-      {
-         _compilationStrategy = new (PERSISTENT_NEW) TR::DefaultCompilationStrategy();
-      }
-
-      if (_compilationStrategy)
-         {
-         TR_OptimizationPlan::_optimizationPlanMonitor = TR::Monitor::create("OptimizationPlanMonitor");
-         _useController = (TR_OptimizationPlan::_optimizationPlanMonitor != 0);
-         if (_useController)
-            {
-            static char *verboseController = feGetEnv("TR_VerboseController");
-            if (verboseController)
-               setVerbose(atoi(verboseController));
-            if (verbose() >= LEVEL1)
-               fprintf(stderr, "Using %s comp strategy\n", strategyName);
-            }
-         }
-      }
-   //TR_ASSERT(_useController, "Must use compilation controller");
-//#ifdef COMP_YIELD_ANALYSIS
-   if (options->getOption(TR_EnableCompYieldStats))
```

Comment:
Is this done in some other part of the code? I don't see it in the omr linked PR.

Reply:
Yes, dependent OMR change is https://github.com/eclipse/omr/pull/6884

Reply:
I don't see where we allocate CompYieldStatsMatrix. Could you please point that out to me (filename and line number)?

Reply:
I don't see either, only appear in J9Compilation [hpp](https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/compile/J9Compilation.hpp#L111) and [cpp](https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/compile/J9Compilation.cpp#L313)

Reply:
In the old implementation this matrix was allocated by the compilation controller. What entity is allocating this data structure now?

Reply:
My bad: I missed it, now I add it back in [OMR](https://github.com/eclipse/omr/commit/32e3dd66941a8949e59e18367ac116cf6c6bc00a) guarded under J9_PROJECT_SPECIFIC
compiler/control/CompilationController.cpp after change at line 57